File: solutions/go/sum-of-multiples/1/sum_of_multiples.go
package summultiples

func SumMultiples(limit int, divisors ...int) int {
	res := 0
    multiples := make(map[int]struct{})
    for _, divisor := range divisors {
        if divisor == 0 {
            continue // Пропускаем делители, равные нулю
        }
        for i := divisor; i < limit; i += divisor {
            multiples[i] = struct{}{}
        }
    }
    for multiple := range multiples {
        res += multiple
    }
	return res
}

File: solutions/python/sum-of-multiples/1/sum_of_multiples.py
def sum_of_multiples(limit, multiples):
    set_res = set()
    res = 0
    for div in multiples:
        if div == 0:
            continue
        for i in range(div, limit, div):
            set_res.add(i)

    for s in set_res:
        res += s
    return res
